src/containers/ChatContainer.tsx

```diff
@@ -1,96 +1,63 @@
-  createGetQueueMessage,
-} from '../services/message-service';
-import { getSocket } from '../utils';
+import React, { useContext, useEffect, useState } from 'react';
+import ChatBodyComponent from '../components/Chat/ChatBodyComponent';
+import ChatHeaderComponent from '../components/Chat/ChatHeaderComponent';
+import ChatInputComponent from '../components/Chat/ChatInputComponent';
 
-const ChatContainer = () => {
-  const [socket, setSocket] = useState(null as any);
-  const [messages, setMessages] = useState<ITextMessage[]>([]);
-  const [roomID, setRoomID] = useState<string>('');
-  const [uniqueID, setUniqueID] = useState<string>('');
-  const [queue, setQueue] = useState<string[]>([]);
-
-  useEffect(() => {
-    setSocket(getSocket());
-  }, []);
-
-  const generateTextMessageFromPayload = (
-    message: ISocketMessage,
-  ): ITextMessage => {
-    return message.payload as ITextMessage;
-  };
-
-  const socketHandler = (message): void => {
-    const parsedMessage: ISocketMessage = JSON.parse(message.data);
-
-    if (parsedMessage.type === 'textMessage') {
-      setMessages(messages => [
-        ...messages,
-        generateTextMessageFromPayload(parsedMessage),
-      ]);
-    } else if (parsedMessage.type === 'distributeRoomMessage') {
-      setRoomID(parsedMessage.payload['roomID']);
-    } else if (parsedMessage.type === 'connectionMessage') {
-      setUniqueID(parsedMessage.payload['uniqueID']);
-    } else if (parsedMessage.type === 'setQueueMessage') {
-      setQueue(parsedMessage.payload['queueMembers']);
-    }
-  };
+import ActiveChatsComponent from '../components/Chat/ActiveChatsComponent';
+import { readMessages } from '../reducers';
+import { SocketContext } from '../providers';
 
-  useEffect(() => {
-    if (!socket) {
-      return;
-    }
-    socket.onmessage = socketHandler;
-  }, [socket]);
+// main component
+const ChatContainer = () => {
+  const [activeIndex, setActiveIndex] = useState<number>(0);
+  const { uniqueID, chats, dispatchChats } = useContext(SocketContext);
 
   useEffect(() => {
     // Auto scroll down in chat
     const display = document.querySelector('.display');
     if (display) {
       display.scrollTo(0, display.scrollHeight);
     }
-  }, [messages]);
+  }, [chats]);
 
-  const onSendTextAndFileMessage = (message: ISocketMessage): void => {
-    setMessages(messages => [
-      ...messages,
-      generateTextMessageFromPayload(message),
-    ]);
-    socket.send(JSON.stringify(message));
+  const showMessages = (index: number) => {
+    setActiveIndex(index);
+    dispatchChats(readMessages(chats[activeIndex].roomID));
   };
 
-  const onSendGetQueueMessage = (): void => {
-    const getMessage: IGetMessage = createGetQueueMessage();
-    socket.send(JSON.stringify(getMessage));
-  };
-
-  const onSendGenerateRoomMessage = (studentID: string): void => {
-    const socketMessage: ISocketMessage = createGenerateRoomMessage(
-      uniqueID,
-      studentID,
+  if (chats.length > 0) {
```

Comment:
Smak og behag, men jeg synes det er enklere å lese når edge-casen med få linjer markup (ingen chats) blir returnert først.

```suggestion
  if (chats.length <= 0) {
```